components/multiselect/multiselect.tsx

```diff
@@ -1,115 +1,146 @@
+    options: Value[];
+    perPage: number;
+    placeholderText?: string;
+    saving?: boolean;
+    submitImmediatelyOn?: (value: Value) => void;
+    totalCount?: number;
+    users?: unknown[];
+    valueRenderer: typeof StateManager;
+    values: Value[];
+}
+
+export type State = {
+    a11yActive: boolean;
+    input: string;
+    page: number;
+}
 
 const KeyCodes = Constants.KeyCodes;
 
-export default class MultiSelect extends React.Component {
-    static propTypes = {
-        users: PropTypes.arrayOf(PropTypes.object),
-        totalCount: PropTypes.number,
-        options: PropTypes.arrayOf(PropTypes.object),
-        optionRenderer: PropTypes.func,
-        values: PropTypes.arrayOf(PropTypes.object),
-        valueRenderer: PropTypes.func,
-        ariaLabelRenderer: PropTypes.func.isRequired,
-        handleInput: PropTypes.func,
-        handleDelete: PropTypes.func,
-        perPage: PropTypes.number,
-        handlePageChange: PropTypes.func,
-        handleAdd: PropTypes.func,
-        handleSubmit: PropTypes.func,
-        noteText: PropTypes.node,
-        maxValues: PropTypes.number,
-        numRemainingText: PropTypes.node,
-        buttonSubmitText: PropTypes.node,
-        buttonSubmitLoadingText: PropTypes.node,
-        submitImmediatelyOn: PropTypes.func,
-        saving: PropTypes.bool,
-        loading: PropTypes.bool,
-        placeholderText: PropTypes.string,
-    }
+export default class MultiSelect extends React.Component<Props, State> {
+    public listRef = React.createRef<MultiSelectList>()
+    public reactSelectRef = React.createRef<ReactSelect>()
+    public selected: Value | null = null
 
-    static defaultProps = {
+    public static defaultProps = {
         ariaLabelRenderer: defaultAriaLabelRenderer,
     }
 
-    constructor(props) {
+    public constructor(props: Props) {
         super(props);
 
-        this.selected = null;
-
         this.state = {
+            a11yActive: false,
             page: 0,
             input: '',
         };
     }
 
-    componentDidMount() {
-        document.addEventListener('keydown', this.handleEnterPress);
-        if (this.refs.reactSelect) {
-            this.refs.reactSelect.select.inputRef.addEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
-            this.refs.reactSelect.select.inputRef.addEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
+    public componentDidMount() {
+        const inputRef: unknown = this.reactSelectRef.current && this.reactSelectRef.current.select.inputRef;
+
+        document.addEventListener<'keydown'>('keydown', this.handleEnterPress);
+        if (inputRef && typeof (inputRef as HTMLElement).addEventListener === 'function') {
+            (inputRef as HTMLElement).addEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
+            (inputRef as HTMLElement).addEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yActivateEvent);
```

Comment:
You have changed it from `handleA11yDeactivateEvent` to `handleA11yActivateEvent`

Reply:
🙀 Thank you for catching this.